src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,273 @@
+#include <mlpack/core/math/random.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/data/string_cleaning.hpp>
+#include <mlpack/core/data/extension.hpp>
+#include <mlpack/core/data/tokenizer/split_by_char.hpp>
+
+PROGRAM_INFO("preprocess_string",
+    // Short description.
+    "A utility to preprocess string data. This utility can remove stopwords, "
+    "punctuation and convert to lowercase.",
+    // Long description.
+    "This utility takes a dataset and the dimension and arguments and "
+    "does the preprocessing of string dataset according to arguments given."
+    "\n\n"
+    "The dataset may be given as the file name and the output may be saved as "
+    + PRINT_PARAM_STRING("actual_dataset") + " and " +
+    PRINT_PARAM_STRING("preprocess_dataset") + " ."
+    "\n\n"
+    " Following arguments may be given " + PRINT_PARAM_STRING("lowercase") +
+    " to convert the dataset to lower case and " +
+    PRINT_PARAM_STRING("punctuation") + "for removing punctuation and "
+    + PRINT_PARAM_STRING("stopwords") + "for removing stopwords. Also the "
+    "dimension which contains the string dataset " +
+    PRINT_PARAM_STRING("dimension") + "."
+    "\n\n"
+    "So, a simple example where we want to preprocess string the dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN("columnDelimiter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d", "\t");
+PARAM_STRING_IN("delimiter", "A set of chars that is used as delimeter to"
+    "tokenize the string dataset ", "D", " ");
+PARAM_STRING_IN("stopwordsfile", "File containing stopwords", "S", "");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_VECTOR_IN_REQ(std::string, "dimension", "Column which contains the"
+    "string data. (1 by default)", "c");
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+static vector<vector<string>> CreateDataset(const string& filename,
+                                            const char& columnDelimiter)
+{
+  vector<vector<string>> dataset;
+  // Extracting the Contents of file
+  // File stream.
+  ifstream fin(filename);
+  if (!fin.is_open())
+    Log::Fatal << "Unable to open input file \n";
```

Comment:
I think each log message should end with `endl` since `\n` doesn't flush the buffer. I didn't point out the same issues below. Could you fix them as well?